### PR TITLE
prepare next release

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

**Comment by phryneas** [comment]
So we'd need to make a new release for the changes from #274 to land in the README. Also this lands a few test improvements and `feat(apiincrementalchecker): improve generation of diagnostics`.

Since the `beta` branch is protected I don't really know how to best prepare merging this into master @piotr-oles, can you help here?

**Comment by piotr-oles** [comment]
I will take a look today’s evening :)

**Comment by piotr-oles** [comment]
I've resolved conflicts in the `CHANGELOG.md` - now everything should be good. I think semantic-release will release new beta version (1.2.1-beta.1).

**Review by piotr-oles — DISMISSED** [review]

**Comment by phryneas** [comment]
Great! Who wants to pull the trigger? :smile:

**Review by johnnyreilly — DISMISSED** [review]
First time with the semantic release hotness!

**Comment by johnnyreilly** [comment]
It looks like we can't merge as Travis isn't running or something... @piotr-oles can you investigate please?

**Review by piotr-oles — DISMISSED** [review]

**Comment by piotr-oles** [comment]
We have all these problems because not all commits were merged into beta :) Travis didn't want to build because when we merged master to beta, semantic-release created new release and updated CHANGELOG.md with `[skip ci]` in the commit message. That why travis didn't want to start build. I have created a new commit with some spacing changes in the CHANGELOG.md to trigger build again without `[skip ci]` in the last commit :)

**Comment by johnnyreilly** [comment]
Cool - is there a way we can avoid this happening in future? I'm keen we have a release process that just works™️ 😊

Is this happening just because this will be our first release end to end with semantic release? So in future these issues won't present? 

I'm a little confused / concerned with the problems we've been facing getting semantic release bedded in. Hopefully we can get past this and into a good release flow 🤞

File: README.md
Comment (swashata) [inline README.md]: Oops for the typo

```suggestion
// These hooks provide access to different events
```

**Review by swashata — CHANGES_REQUESTED** [review]
I have found a typo in the docs. Please accept the changes in this PR too :)

**Comment by piotr-oles** [comment]
The most problematic is generation of the `CHANGELOG.md` - I think that's why it's not in the default semantic-release setup. After we merge to the `master` it creates additional commit with updated `CHANGELOG.md`. This commit doesn't appear in the `beta` branch and this creates merge conflicts in the next pull request (as we modify `CHANGELOG.md` in every beta release). We can manualy merge the master barnch into the beta after release but maybe we could automate this also (that was the point of the semantic-release)

**Comment by johnnyreilly** [comment]
My take away from that is that semantic release doesn't have a good workflow with regards to changelogs. That's super disappointing. 

I'm rather hesitant about us creating our own automation around this as that may end up being another thing to maintain as semantic release evolves. If there's not a good out of the box experience that we can have I'm inclined to say "well nevermind - let's just have the GitHub releases instead"

It means we have to be online to see them and if we were ever to move the plugin elsewhere we'd need to recreate manually. Neither of those is a deal-breaker in my eyes.

I value a simple release workflow above other concerns. If you want to drop the changelog generation I'm fine with that. 😊

**Review by piotr-oles — DISMISSED** [review]

**Comment by phryneas** [comment]
Since the CHANGELOG is automatically generated from commit messages, I think we won't lose any metadata when dropping the automatic generation - even if this project were to move away from github some day in the future. The commit history won't be lost.

**Comment by piotr-oles** [comment]
I've created PR to drop `CHANGELOG.md` generation and remove emoji as not everyone is using them 😢

**Comment by piotr-oles** [comment]
The last thing we should remember is to use "Rebase and merge" when merging from the `beta` to `master` - that will not create additional commit and will create a tag with the new version on the last commit that was also on the beta branch. Thanks to that master and beta will be in sync (otherwise master will be 1 commit ahead of beta - the merge commit). I will create PR with docs update for the deployment process.

**Comment by johnnyreilly** [comment]
Good catch - thanks!

**Review by piotr-oles — APPROVED** [review]

**Comment by johnnyreilly** [comment]
How did you get it to work? Can you just not rebase on a phone?

**Comment by phryneas** [comment]
Nope, looks like this was one last normal merge. I guess we'll have to rebase that one back to beta one last time and then we'll be golden as long as we don't push to master.

**Pull request merged**
